docs: clarify `usePathname` usage as client hook. (#51146)
vercel/next.js: docs/02-app/02-api-reference/04-functions/use-pathname.mdx

}
```

`usePathname` intentionally requires using a [Client Component](/docs/getting-started/react-essentials). It's important to note Client Components are not a de-optimization. They are an integral part of the [Server Components](/docs/getting-started/react-essentials) architecture.

> For example, a Client Component with `usePathname` will be rendered into HTML on the initial page load. When navigating to a new route, this component does not need to be refetched. Instead, the component can be downloaded once (in the client JavaScript bundle) and update what it renders based on the current state.

Reviewer: ```suggestion
For example, a Client Component with `usePathname` will be rendered into HTML on the initial page load. When navigating to a new route, this component does not need to be re-fetched. Instead, the component is downloaded once (in the client JavaScript bundle), and re-renders based on the current state.
```
